### PR TITLE
Delete `docs/`, link to docs.konghq.com instead

### PR DESCRIPTION
**Pull request by mflendrich**
**What this PR does / why we need it**:

- ~Deletes `docs/`.~ As of https://github.com/Kong/docs.konghq.com/pull/2446, docs.konghq.com is "the" place for KIC docs.
    - Per https://github.com/Kong/kubernetes-ingress-controller/pull/966#issuecomment-735876549, this PR replaces content with a link to docs.konghq.com instead of deleting files.
- Updates `docs/` links to point to equivalent documents at docs.konghq.com.

**Special notes for your reviewer**:
Some PRs (#963, #961, #952) to `docs/` have been merged after https://github.com/Kong/docs.konghq.com/pull/2446, porting them will follow.

Some PRs are blocked by this PR: #954

### CODE REVIEW

**Comment by hbagdi** [comment]
This change will result in the horrendously useless 404 Github page ([like this](https://github.com/Kong/kubernetes-ingress-controller/blob/main/docs/broken-on-purpose.md)) for all the pages. A lot of these pages are bookmarked or hrefed from multiple sites. An alternative here is to update the content on every page to redirect them to the docs site. This is certainly a little bit more work but leads to much better user experience.

**Comment by mflendrich** [comment]
> An alternative here is to update the content on every page to redirect them to the docs site. This is certainly a little bit more work but leads to much better user experience.

That makes sense to me. Will do and ping.

Adding @lena-larionova as a reviewer, too.

**Review by hbagdi — DISMISSED** [review]
LGTM

**Comment by hbagdi** [comment]
1. There is one conflict.
2. While not part of this PR, we need to update README.md and change links to docs.konghq.com for all documentation pointers in that file.

File: deploy/README.md [hunk 1]
```diff
@@ -1,3 +1,3 @@
 # 301 Moved
 
-This document is now moved to [Deployment Guides](../docs/deployment).
\ No newline at end of file
+This document is now moved to [Deployment Overwiew on `docs.konghq.com`](https://docs.konghq.com/kubernetes-ingress-controller/latest/deployment/overview/).
```
Comment (lena-larionova) [inline deploy/README.md:3]:
```suggestion
This document is now moved to [Deployment Overview on `docs.konghq.com`](https://docs.konghq.com/kubernetes-ingress-controller/latest/deployment/overview/).
```

**Review by lena-larionova — APPROVED** [review]
Just one typo, otherwise LGTM.

**Comment by mflendrich** [comment]
@hbagdi 


> There is one conflict.

Resolved.


> While not part of this PR, we need to update README.md and change links to docs.konghq.com for all documentation pointers in that file.

Maybe I misunderstand something, but I think that this PR _does_ address that already - I think it updates all links this way. PTAL

**Review by hbagdi — APPROVED** [review]

**Pull request merged**